### PR TITLE
chore: add support for trino 455, remove old opa authorizer code

### PR DESCRIPTION
**Pull request by xeniape**
# Description

Part of https://github.com/stackabletech/issues/issues/626

- Add support for Trino version `455`
- Remove support for Trino versions `414` and `442`
- Remove OPA authorizer code for older Trino versions

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes

Tests passed
```
--- PASS: kuttl (1447.13s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/orphaned-resources_trino-latest-455_openshift-false (93.22s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-false_use-tls-true_use-internal-tls-false_openshift-false (123.55s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-true_use-tls-true_use-internal-tls-true_openshift-false (126.78s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-true_use-tls-true_use-internal-tls-false_openshift-false (91.05s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-true_use-tls-false_use-internal-tls-false_openshift-false (68.24s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-true_use-tls-false_use-internal-tls-true_openshift-false (76.04s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-false_use-tls-false_use-internal-tls-false_openshift-false (70.04s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-false_use-tls-true_use-internal-tls-true_openshift-false (85.64s)
        --- PASS: kuttl/harness/tls_trino-latest-455_use-authentication-false_use-tls-false_use-internal-tls-true_openshift-false (74.96s)
        --- PASS: kuttl/harness/smoke_trino-455_hive-4.0.0_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-false_openshift-false (418.97s)
        --- PASS: kuttl/harness/smoke_trino-451_hive-3.1.3_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-false_openshift-false (372.61s)
        --- PASS: kuttl/harness/smoke_trino-451_hive-3.1.3_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-true_openshift-false (442.21s)
        --- PASS: kuttl/harness/smoke_trino-455_hive-3.1.3_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-false_openshift-false (445.15s)
        --- PASS: kuttl/harness/cluster-operation_trino-latest-455_openshift-false (103.28s)
        --- PASS: kuttl/harness/smoke_trino-455_hive-3.1.3_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-true_openshift-false (432.06s)
        --- PASS: kuttl/harness/authentication_trino-latest-455_ldap-use-tls-true_openshift-false (308.13s)
        --- PASS: kuttl/harness/resources_trino-latest-455_openshift-false (55.64s)
        --- PASS: kuttl/harness/opa-authorization_trino-latest-455_hive-latest-4.0.0_opa-0.66.0_keycloak-25.0.0_openshift-false (322.12s)
        --- PASS: kuttl/harness/smoke_trino-451_hive-4.0.0_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-false_openshift-false (362.97s)
        --- PASS: kuttl/harness/smoke_trino-451_hive-4.0.0_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-true_openshift-false (391.36s)
        --- PASS: kuttl/harness/logging_trino-455_openshift-false (94.47s)
        --- PASS: kuttl/harness/authentication_trino-latest-455_ldap-use-tls-false_openshift-false (290.56s)
        --- PASS: kuttl/harness/logging_trino-451_openshift-false (86.77s)
        --- PASS: kuttl/harness/smoke_trino-455_hive-4.0.0_opa-0.66.0_hdfs-3.4.0_zookeeper-3.9.2_s3-use-tls-true_openshift-false (405.29s)
PASS
```

```[tasklist]
# Dependencies
- [ ] https://github.com/stackabletech/docker-images/pull/822
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [x] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs/style-guide).
- [x] Changelog updated
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: rust/operator-binary/src/config/jvm.rs
Comment (maltesander) [inline rust/operator-binary/src/config/jvm.rs:96]: The config https://trino.io/docs/455/installation/deployment.html#jvm-config is a bit different from https://trino.io/docs/452/installation/deployment.html#jvm-config, did you check that?

**Review by maltesander — COMMENTED** [review]
Thanks, please update the supported versions in `docs/modules/trino/partials/supported-versions.adoc` as well.

File: rust/operator-binary/src/config/jvm.rs
Comment (xeniape) [reply rust/operator-binary/src/config/jvm.rs:96]: Sorry the PR wasn't ready for review, still working on it :laughing: that's why I didn't move it to `waiting for review` yet. `supported-versions.adoc` is updated in the commit I'm working on right now.

The config removed the settings I assumed were a workaround for the linked fix. But this fix is available in openjdk-23, which we can't use yet? https://pkgs.org/download/java-23-openjdk-devel but if you have more information let me know.

Comment (maltesander) [reply rust/operator-binary/src/config/jvm.rs:96]: Ok nvm its the `Draft` vs `In review` thing :)
I did not look further into the jvm config, just wanted to point out that something changed.

Comment (xeniape) [reply rust/operator-binary/src/config/jvm.rs:96]: Wasn't sure if I would get a PR number for the changelog with a Draft :D will try it out next time, sorry for the confusion

Comment (maltesander) [reply rust/operator-binary/src/config/jvm.rs:96]: No worries, you did it right, its about whats "in review" :D so my bad...

**Review by maltesander — APPROVED** [review]
LGTM!

**Review by sbernauer — APPROVED** [review]
LGTM when tests pass

File: rust/operator-binary/src/config/jvm.rs
Comment (xeniape) [reply rust/operator-binary/src/config/jvm.rs:96]: Checked with @sbernauer regarding the jvm settings: https://bugs.openjdk.org/browse/JDK-8330772 ; removed it

**Pull request merged**
